### ISSUE TITLE
Harmonize sees the Hue Hub but cannot sync + NameError: name 'baseurl' is not defined

### ISSUE THREAD

**Issue by MCPCapital** [MCPCapital/HarmonizeProject#7]
Opened by @bboy486 on August 13th

This is two issues in one. One is fixed.

1- The project sees the hue hub and when I press the button (or hold it down) I get an error that it has timed out have a nice day (something like that). I had to
cd harmonizeproject
and install from the folder.

2- After I did that I was able to add my hub. But I did receive the error below.
Success! I generated a username and client key to access the bridge's Entertainment API!
Traceback (most recent call last):
File "./harmonize.py", line 143, in
r = requests.get(url = baseurl+"/config")
NameError: name 'baseurl' is not defined

**Comment by MCPCapital** [comment]
@SeattleSi and @evangryz confirmed seeing the second error listed.

On August 14h SeattleSi commented:

I'm seeing issue #2 as well.

Traceback (most recent call last):
File "./harmonizeproject/harmonize.py", line 143, in
r = requests.get(url = baseurl+"/config")
NameError: name 'baseurl' is not defined

However, after restarting ./harmonizeproject/harmonize.py I'm seeing:

Press return to stopException in thread Thread-2:
Traceback (most recent call last):
File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
self.run()
File "/usr/lib/python3.7/threading.py", line 865, in run
self._target(*self._args, **self._kwargs)
File "./harmonizeproject/harmonize.py", line 226, in averageimage
coords[0] = ((coords[0])+1) * w//2 #Translates x value and resizes to video aspect ratio
NameError: name 'w' is not defined

Exception in thread Thread-3:
Traceback (most recent call last):
File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
self.run()
File "/usr/lib/python3.7/threading.py", line 865, in run
self._target(*self._args, **self._kwargs)
File "./harmonizeproject/harmonize.py", line 309, in buffer_to_light
for i in rgb_bytes:
NameError: name 'rgb_bytes' is not defined

**Comment by MCPCapital** [comment]
On August 17th I replied:
For your # 1, I think this is the expected behavior. You gotta install the dependencies it before you can run it! I think it also generates client.json where your terminal session 'is'; but then always looks for it in the root directory. We can update the instructions to clarify you must cd into the directory first, thanks!

For your # 2, I can see this bug in the code now. The issue is we only define 'baseurl' IF client.json is found. If you haven't set it up yet, there is no client.json and thus you see the error you saw. I think it should run on the second try though, as SeattleSi pointed out. I'll label this a bug and get to fixing it as well.

Regarding SeattleSi 's issue, first off, try again. Sometimes one thread gets ahead of another and doesn't define the video shape in time. If that doesn't work, you may have an issue getting video input. Does the videotest.py script work for you?

**Comment by MCPCapital** [comment]
On August 18th I closed the issue with:

I have some updates on this.

Sub-Issue # 1 has been clarified in the Readme, telling users they must CD into the directory to generate the clientdata properly.

To clarify the issue with # 2 in this issue - it is that the below code which defines the baseurl is only triggered if Path("./client.json").is_file(): {See line 122}. To fix this we need to include the below code in the 'else' tied to the same IF statement

```
    global baseurl
    baseurl = "http://{}/api".format(hueip)
```

For now I've included the workaround instructions in the Readme, but I will fix this in the code sometime soon. Thanks again for calling these out!
